cpp/src/io/json/json_tree.cu

```diff
@@ -402,20 +419,21 @@ rmm::device_uvector<size_type> hash_node_type_with_field_name(device_span<Symbol
       d_input + node_range_begin[node_id2], node_range_end[node_id2] - node_range_begin[node_id2]);
     return field_name1 == field_name2;
   };
-  auto is_field_name_node = [node_categories = d_tree.node_categories.data()] __device__(
-                              auto node_id) { return node_categories[node_id] == node_t::NC_FN; };
   // key-value pairs: uses node_id itself as node_type. (unique node_id for a field name due to
   // hashing)
   auto iter = cudf::detail::make_counting_transform_iterator(
     0, [] __device__(size_type i) { return cuco::make_pair(i, i); });
 
+  auto is_field_name_node = [node_categories = d_tree.node_categories.data()] __device__(
```

Comment:
`const` if possible.

Reply:
should lambda's be const? does it make any difference?

Reply:
Lambda is just a type of variable. So we can (and should) make it `const`.